### ISSUE TITLE
[clang-tidy][false-positive] `modernize-use-nullptr` suggests comparing `std::strong_ordering` with `nullptr`

### ISSUE THREAD

**Issue by tearfur** [llvm/llvm-project#206245]
`std::strong_ordering::operator==` is only defined for literal zero `0`. However, `modernize-use-nullptr` suggests to replace `0` with `nullptr`.

https://godbolt.org/z/ME9Gs3Pb5

```
<source>:4:41: warning: use nullptr [modernize-use-nullptr]
    4 |     void(std::strong_ordering::equal == 0);
      |                                         ^
      |                                         nullptr
<source>:5:46: warning: use nullptr [modernize-use-nullptr]
    5 |     void(std::strong_ordering::equivalent == 0);
      |                                              ^
      |                                              nullptr
<source>:6:40: warning: use nullptr [modernize-use-nullptr]
    6 |     void(std::strong_ordering::less == 0);
      |                                        ^
      |                                        nullptr
<source>:7:43: warning: use nullptr [modernize-use-nullptr]
    7 |     void(std::strong_ordering::greater == 0);
      |                                           ^
      |                                           nullptr
4 warnings generated.
```

Current trunk: 5e47ef456a7860e1fa12428f96ff072ff6c4f905
Possibly related to #53961.
